src/Foundation/FileStorage.php

```diff
@@ -39,6 +39,6 @@ public function set(string $key, $value): bool
 
     public function delete(string $key): bool
     {
-        return unlink($this->path($key));
+        return file_exists($path = $this->path($key)) && unlink($path);
```

Comment:
No need for `file_exists` check here. Only `unlink` works fine. I'm OK with preserving warning, in case that's your goal, just use `@unlink` then.